lbz/authz/wrappers.py

```diff
@@ -0,0 +1,70 @@
+# coding=utf-8
+"""
+Authorization helper wrappers (decorators).
+"""
+from functools import wraps
+from typing import Callable, Any
+
+from lbz.authz.authorizer import Authorizer
+from lbz.authz.collector import AuthzCollector
+from lbz.exceptions import PermissionDenied, Unauthorized
+from lbz.resource import Resource
+
+
+def check_permission(resource: Resource, permission_name: str) -> dict:
+    """
+    Check if requester has sufficient permissions to do something on specific resource.
+
+    Raises if not.
+    """
+    authorization_header = resource.request.headers.get("Authorization", "")
+    guest_authorization_policy = resource.get_guest_authorization()
+    if not authorization_header and not guest_authorization_policy:
+        raise Unauthorized("Authorization header missing or empty")
+
+    authorizer = Authorizer(
+        auth_jwt=authorization_header,
+        resource_name=resource.get_name(),
+        permission_name=permission_name,
+        policy_override=guest_authorization_policy,
+    )
+    authorizer.check_access()
+    return authorizer.restrictions
+
+
+def has_permission(resource: Resource, permission_name: str) -> bool:
+    """
+    Safe Check if requester has sufficient permissions to do something on specific resource.
+
+    Does not raise.
+    """
+    try:
+        check_permission(resource, permission_name)
+    except (Unauthorized, PermissionDenied):
+        return False
+    return True
+
+
+authz_collector = AuthzCollector()
```

Comment:
Why not initialize it inside the `collector.py`? It sounds reasonable to have it there instead of keeping it here between functions 🤔